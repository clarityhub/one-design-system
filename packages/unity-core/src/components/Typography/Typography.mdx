---
name: Typography
menu: Theme
---

import { Playground, PropsTable } from 'docz';
import Typography from './Typography';

# Typography

## \<Typography /\> component

```js
import Typography from '@clarityhub/unity-core/components/Typography';
```

## Prop Types

<PropsTable of={Typography} />

## Examples

### Text versus Paragraph

By default, if no `type` is provided to `Typography`, then `text` is used.

If you want to have multiple paragraphs of readable text, use `type="p"`.

<Playground>
    <Typography>Lorem ipsum dolor sit amet, <i>consectetur adipiscing elit</i>. Donec odio ipsum, <strong>sodales et ligula sed</strong>, venenatis cursus libero. Suspendisse sit amet tempus tortor, tincidunt egestas tortor. In hac habitasse platea dictumst. Sed consectetur nulla eget diam tincidunt eleifend.</Typography>
    <Typography type="p">Lorem ipsum dolor sit amet, <i>consectetur adipiscing elit</i>. Donec odio ipsum, <strong>sodales et ligula sed</strong>, venenatis cursus libero. Suspendisse sit amet tempus tortor, tincidunt egestas tortor. In hac habitasse platea dictumst. Sed consectetur nulla eget diam tincidunt eleifend.</Typography>
</Playground>

### Headings

<Playground>
    <Typography type="h1">H1. The quick brown fox</Typography>
    <Typography type="h2">H2. The quick brown fox</Typography>
    <Typography type="h3">H3. The quick brown fox</Typography>
    <Typography type="h4">H4. The quick brown fox</Typography>
    <Typography type="h5">H5. The quick brown fox</Typography>
</Playground>

Without padding:

<Playground>
    <Typography type="h1" noPadding>H1. The quick brown fox</Typography>
    <Typography type="h2" noPadding>H2. The quick brown fox</Typography>
    <Typography type="h3" noPadding>H3. The quick brown fox</Typography>
    <Typography type="h4" noPadding>H4. The quick brown fox</Typography>
    <Typography type="h5" noPadding>H5. The quick brown fox</Typography>
</Playground>

### Text

<Playground>
    <Typography type="p">Lorem ipsum dolor sit amet, <i>consectetur adipiscing elit</i>. Donec odio ipsum, <em>sodales et ligula sed</em>, venenatis cursus libero. Suspendisse sit amet tempus tortor, tincidunt egestas tortor. In hac habitasse platea dictumst. Sed consectetur nulla eget diam tincidunt eleifend.</Typography>
    <Typography type="p">Vestibulum quis facilisis dolor, <strong>a faucibus urna</strong>. Vestibulum lobortis massa in <b>lacus semper</b>, sit amet ultrices eros lacinia. Aenean consectetur luctus est, ut tincidunt arcu hendrerit eu. Aliquam orci dui, iaculis vitae vestibulum sed, ullamcorper ut odio. Maecenas ut mattis nisi. Fusce in ante sit amet magna ullamcorper euismod id ut lacus. Duis eget auctor velit.</Typography>
    <Typography type="subtext">This is a set of subtext</Typography>
    <Typography type="sectionLabel">This is a section label</Typography>
</Playground>

### Kitchen Sink

<Playground>
    <Typography type="h1">H1. The quick brown fox</Typography>
    <Typography type="p">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec odio ipsum, sodales et ligula sed, venenatis cursus libero. Suspendisse sit amet tempus tortor, tincidunt egestas tortor. In hac habitasse platea dictumst. Sed consectetur nulla eget diam tincidunt eleifend.</Typography>
    <Typography type="p">Vestibulum quis facilisis dolor, a faucibus urna. Vestibulum lobortis massa in lacus semper, sit amet ultrices eros lacinia. Aenean consectetur luctus est, ut tincidunt arcu hendrerit eu. Aliquam orci dui, iaculis vitae vestibulum sed, ullamcorper ut odio. Maecenas ut mattis nisi. Fusce in ante sit amet magna ullamcorper euismod id ut lacus. Duis eget auctor velit.</Typography>

    <Typography type="h2">H2. The quick brown fox</Typography>

    <Typography type="p">Vestibulum quis facilisis dolor, a faucibus urna. Vestibulum lobortis massa in lacus semper, sit amet ultrices eros lacinia. Aenean consectetur luctus est, ut tincidunt arcu hendrerit eu. Aliquam orci dui, iaculis vitae vestibulum sed, ullamcorper ut odio. Maecenas ut mattis nisi. Fusce in ante sit amet magna ullamcorper euismod id ut lacus. Duis eget auctor velit.</Typography>

    <Typography type="h2">H2. The quick brown fox</Typography>

    <Typography type="p">Vestibulum quis facilisis dolor, a faucibus urna. Vestibulum lobortis massa in lacus semper, sit amet ultrices eros lacinia. Aenean consectetur luctus est, ut tincidunt arcu hendrerit eu. Aliquam orci dui, iaculis vitae vestibulum sed, ullamcorper ut odio. Maecenas ut mattis nisi. Fusce in ante sit amet magna ullamcorper euismod id ut lacus. Duis eget auctor velit.</Typography>
</Playground>