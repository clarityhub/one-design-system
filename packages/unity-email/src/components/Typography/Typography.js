import React from 'react';
import {
	MjmlText,
} from 'mjml-react';
import colors from '@clarityhub/unity-core/dist/theme/colors';

import { types, variants } from '../../theme/fonts';

const Typography = ({
	type = 'text',
	variant,
	color = 'dark',
	children,
	component: suggestedComponent = 'p',
	noMargin = false,
	noPadding = false,
	inline = false,
	css: userCss,
	center = false,
	...rest
}) => {
	const props = Object.assign(
		{},
		Boolean(type) ? types[type] : {},
		Boolean(variant) ? variants[variant] : {},
		Boolean(color) ? { color: colors[color] } : {},
		center ? { align: 'center' } : { align: 'left' },
	);

	return <MjmlText {...rest} {...props}>{children}</MjmlText>;
};

// XXX prop types

export default Typography;
