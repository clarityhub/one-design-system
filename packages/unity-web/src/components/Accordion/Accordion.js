import React, { useState, useCallback } from 'react';
import styled from '@emotion/styled';
import { bool, func, node, oneOfType } from 'prop-types';

const AccordionStyled = styled.div`
`;

const Accordion = ({ multi, children, ...props }) => {
	const [openItems, setOpenItems] = useState([]);

	const onExpand = useCallback((i) => {
		if (multi) {
			setOpenItems((openItems) => {
				return {
					...openItems,
					[i]: true,
				};
			});
		} else {
			setOpenItems((openItems) => {
				return {
					[i]: true,
				};
			});
		}
	}, []);

	const onClose = useCallback((i) => {
		setOpenItems((openItems) => {
			return {
				...openItems,
				[i]: false,
			};
		});
	}, []);

	return (
		<AccordionStyled {...props}>
			{React.Children.map(children, (child, index, arr) => {
				return React.cloneElement(child, {
					expanded: openItems[index],
					onExpand,
					onClose,
					index,
				});
			})}
		</AccordionStyled>
	);
};

Accordion.propTypes = {
	children: oneOfType([func, node]).isRequired,
	multi: bool,
};

Accordion.defaultProps = {
	multi: false,
};

export default Accordion;