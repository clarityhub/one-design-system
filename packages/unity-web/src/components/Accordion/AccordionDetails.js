import React, { useRef, useLayoutEffect, useState } from 'react';
import { bool, func, node, oneOfType } from 'prop-types';
import styled from '@emotion/styled';
import { css } from '@emotion/core';

import { CardBody } from '../Card';

const AccordionDetailsStyled = styled.div`
    overflow: hidden;
    transition: max-height 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;

    ${({ expanded, predictedHeight }) => expanded ? css`
        max-height: ${predictedHeight}px;
    ` : css`
        max-height: 0;
    `}
`;

const AccordionDetails = ({ expanded, children, ...props }) => {
	const ref = useRef();
	const [height, setHeight] = useState(0);

	useLayoutEffect(() => {
		const h = ref.current.scrollHeight;

		setHeight(h);
	}, []);

	return (
		<AccordionDetailsStyled ref={ref} predictedHeight={height} expanded={expanded} {...props}>
			<CardBody>
				{children}
			</CardBody>
		</AccordionDetailsStyled>
	);
};

AccordionDetails.propTypes = {
	children: oneOfType([func, node]).isRequired,
	expanded: bool,
};

AccordionDetails.defaultProps = {
	expanded: false,
};

export default AccordionDetails;