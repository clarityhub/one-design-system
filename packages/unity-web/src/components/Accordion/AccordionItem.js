import React from 'react';
import { bool, func, node, number, oneOfType } from 'prop-types';
import styled from '@emotion/styled';
import { css } from '@emotion/core';
import borders from '@clarityhub/unity-core/lib/borders';

import Card from '../Card';

const AccordionItemStyled = styled(Card)`
	margin-bottom: 0;
	transition: all 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
	transition-property: margin-top, margin-bottom;

	border-bottom-left-radius: 0;
	border-bottom-right-radius: 0;
	border-top-left-radius: 0;
	border-top-right-radius: 0;

	&:first-of-type {
		border-top-left-radius: ${borders.borderRadius.rem}rem !important;
		border-top-right-radius: ${borders.borderRadius.rem}rem !important;
	}

	&:last-of-type {
		border-bottom-left-radius: ${borders.borderRadius.rem}rem !important;
		border-bottom-right-radius: ${borders.borderRadius.rem}rem !important;
	}

	${({ expanded }) => expanded && css`
		border-top-left-radius: ${borders.borderRadius.rem}rem !important;
		border-top-right-radius: ${borders.borderRadius.rem}rem !important;
		border-bottom-left-radius: ${borders.borderRadius.rem}rem !important;
		border-bottom-right-radius: ${borders.borderRadius.rem}rem !important;

		&:not(:first-of-type) {
			margin-top: 1rem;
		}

		&:not(:last-of-type) {
			margin-bottom: 1rem;
		}
	`}
`;

const AccordionItem = ({ expanded, onExpand, onClose, index, children, ...props }) => {
	return (
		<AccordionItemStyled expanded={expanded} {...props}>
			{React.Children.map(children, (child) => {
				return React.cloneElement(child, {
					expanded,
					onExpand,
					onClose,
					index,
				});
			})}
		</AccordionItemStyled>
	);
};

AccordionItem.propTypes = {
	children: oneOfType([func, node]).isRequired,
	expanded: bool,
	/**
	 * Auto-provided by <Accordion />
	 */
	index: number,
	/**
	 * Auto-provided by <Accordion />
	 */
	onClose: func,
	/**
	 * Auto-provided by <Accordion />
	 */
	onExpand: func,
};

export default AccordionItem;