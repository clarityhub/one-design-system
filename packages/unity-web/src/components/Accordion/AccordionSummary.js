import React, { useCallback } from 'react';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { bool, number, oneOfType, node, func } from 'prop-types';
import spacing from '@clarityhub/unity-core/lib/spacing';
import colors from '@clarityhub/unity-core/lib/colors';

const AccordionSummaryButton = styled.button`
    background-color: transparent;
    border: 0;
    cursor: pointer;
    display: flex;
	outline: none;
    width: 100%;

    padding: 0.75rem ${spacing.small}rem;
    display: flex;
    flex-grow: 1;
    transition: padding 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;

	${({ isDisabled }) => isDisabled && css`
		background-color: ${colors.muted.default};
	`}

	${({ expanded }) => expanded && css`
		padding: 1.25rem ${spacing.small}rem;
	`}
`;

const AccordionSummaryStyled = styled.div`
    flex: 1;
    align-content: center;
    justify-content: left;
    text-align: left;
`;

const IconWrapper = styled.div`
    display: inline-block;
	transform: rotate(0deg);
	transition: transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;

	${({ expanded }) => expanded && css`
		transform: rotate(-180deg);
	`}
`;

const AccordionSummary = ({
	disabled,
	expanded,
	index,
	onClose,
	onExpand,
	children,
	expandIcon,
	onClick,
	...props
}) => {
	const handleClick = useCallback((e) => {
		e.preventDefault();

		if (disabled) {
			return;
		}

		if (expanded) {
			onClose(index);
		} else {
			onExpand(index);
		}

		if (onClick) {
			onClick(e, expanded);
		}
	});

	return (
		<AccordionSummaryButton
			onClick={handleClick}
			isDisabled={disabled}
			disabled={disabled}
			expanded={expanded}
		>
			<AccordionSummaryStyled
				{...props}
			>
				{children}
			</AccordionSummaryStyled>
            
			{expandIcon && (
				<IconWrapper expanded={expanded}>
					{expandIcon}
				</IconWrapper>
			)}
		</AccordionSummaryButton>
	);
};

AccordionSummary.propTypes = {
	children: oneOfType([node, func]).isRequired,
	disabled: bool,
	expandIcon: node,
	/**
	 * Auto-provided by <AccordionItem />
	 */
	expanded: bool,
	/**
	 * Auto-provided by <AccordionItem />
	 */
	index: number,
	onClick: func,
	/**
	 * Auto-provided by <AccordionItem />
	 */
	onClose: func,
	/**
	 * Auto-provided by <AccordionItem />
	 */
	onExpand: func,
};

AccordionSummary.defaultProps = {
	disabled: false,
	node: false,
};

export default AccordionSummary;