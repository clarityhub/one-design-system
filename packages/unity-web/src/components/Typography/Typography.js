/** @jsx jsx */

import { bool, node, oneOf, string } from 'prop-types';
import styled from '@emotion/styled';
import { jsx } from '@emotion/core';
import colors from '@clarityhub/unity-core/lib/colors';
import { variants } from '@clarityhub/unity-core/lib/typography';

import { types } from '../../theme/fonts';

const noPaddingCss = `
    padding: 0;
`;

const noMarginCss = `
    margin-bottom: 0;
    margin-top: 0;

    & + & {
        margin-bottom: 0;
        margin-top: 0;
    }
`;

const Typography = ({
	type = 'text',
	variant,
	color = 'dark',
	children,
	component: suggestedComponent = 'p',
	noMargin = false,
	noPadding = false,
	inline = false,
	css: userCss,
	center = false,
	...rest
}) => {
	let styles = [
		Boolean(type) ? types[type]: {},
		Boolean(variant) ? variants[variant]: {},
		Boolean(color) ? { color: colors[color].default }: {},
		Boolean(noPadding) ? noPaddingCss: {},
		Boolean(noMargin) ? noMarginCss: {},
		Boolean(userCss) ? userCss: {},
		Boolean(inline) ? {
			display: 'inline-block',
		} : {},
		Boolean(center) ? {
			textAlign: 'center',
		} : {},
	];

	if (typeof suggestedComponent === 'string') {
		let component = suggestedComponent;
		if (suggestedComponent === 'p' && ['h1', 'h2', 'h3', 'h4', 'h5'].indexOf(type) !== -1) {
			component = type;
		}

		const Comp = styled[component](styles);
		return <Comp {...rest}>{children}</Comp>;
	} else {
		const Comp = suggestedComponent;
		return <Comp {...rest} css={styles}>{children}</Comp>;
	}
};

Typography.propTypes = {
	center: bool,
	children: node,
	color: oneOf(Object.keys(colors)),
	component: node,
	css: string,
	inline: bool,
	noMargin: bool,
	noPadding: bool,
	type: oneOf(Object.keys(types)),
	variant: oneOf(Object.keys(variants)),
};

export default Typography;
