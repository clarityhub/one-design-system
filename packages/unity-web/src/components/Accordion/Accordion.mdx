---
name: Accordion
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Accordion from './Accordion';
import AccordionItem from './AccordionItem';
import AccordionSummary from './AccordionSummary';
import AccordionDetails from './AccordionDetails';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Typography from '../Typography';

# Accordion

Use accordions to show detailed information to a user.

Don't nest accordions.

```js
import { Accordion, AccordionItem, AccordionSummary, AccordionDetails } from '@clarityhub/unity-web/lib/components/Accordion';
```

## Accordions

<Playground>
    <Accordion>
        <AccordionItem>
            <AccordionSummary
                expandIcon={<ExpandMoreIcon />}
            >
                <Typography inline noMargin noPadding type="text" style={{ marginRight: '30px' }}><b>General settings</b></Typography>
                <Typography inline noMargin noPadding type="text">I am an expansion panel</Typography>
            </AccordionSummary>

            <AccordionDetails>
                <Typography>
                    Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget
                    maximus est, id dignissim quam.
                </Typography>
            </AccordionDetails>
        </AccordionItem>
        <AccordionItem>
            <AccordionSummary
                expandIcon={<ExpandMoreIcon />}
            >
                <Typography inline noMargin noPadding type="text" style={{ marginRight: '30px' }}><b>General settings</b></Typography>
                <Typography inline noMargin noPadding type="text">I am an expansion panel</Typography>
            </AccordionSummary>

            <AccordionDetails>
                <Typography>
                    Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget
                    maximus est, id dignissim quam.
                </Typography>
            </AccordionDetails>
        </AccordionItem>
        <AccordionItem>
            <AccordionSummary
                expandIcon={<ExpandMoreIcon />}
            >
                <Typography inline noMargin noPadding type="text" style={{ marginRight: '30px' }}><b>General settings</b></Typography>
                <Typography inline noMargin noPadding type="text">I am an expansion panel</Typography>
            </AccordionSummary>

            <AccordionDetails>
                <Typography>
                    Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget
                    maximus est, id dignissim quam.
                </Typography>
            </AccordionDetails>
        </AccordionItem>
    </Accordion>
</Playground>

## Disabled and Multi

<Playground>
    <Accordion multi>
        <AccordionItem>
            <AccordionSummary
                disabled
                expandIcon={<ExpandMoreIcon />}
            >
                <Typography inline noMargin noPadding type="text" style={{ marginRight: '30px' }}><b>General settings</b></Typography>
                <Typography inline noMargin noPadding type="text">I am an expansion panel</Typography>
            </AccordionSummary>

            <AccordionDetails>
                <Typography>
                    Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget
                    maximus est, id dignissim quam.
                </Typography>
            </AccordionDetails>
        </AccordionItem>
        <AccordionItem>
            <AccordionSummary
                expandIcon={<ExpandMoreIcon />}
            >
                <Typography inline noMargin noPadding type="text" style={{ marginRight: '30px' }}><b>General settings</b></Typography>
                <Typography inline noMargin noPadding type="text">I am an expansion panel</Typography>
            </AccordionSummary>

            <AccordionDetails>
                <Typography>
                    Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget
                    maximus est, id dignissim quam.
                </Typography>
            </AccordionDetails>
        </AccordionItem>
        <AccordionItem>
            <AccordionSummary
                expandIcon={<ExpandMoreIcon />}
            >
                <Typography inline noMargin noPadding type="text" style={{ marginRight: '30px' }}><b>General settings</b></Typography>
                <Typography inline noMargin noPadding type="text">I am an expansion panel</Typography>
            </AccordionSummary>

            <AccordionDetails>
                <Typography>
                    Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget
                    maximus est, id dignissim quam.
                </Typography>
            </AccordionDetails>
        </AccordionItem>
    </Accordion>
</Playground>


## Prop Types

### Accordion

The following props are available for the Accordion component.

The `<Accordion />` component will pass down `expanded`, `index`, `onExpand`, `onClose` are passed down
to `AccordionItem`, `AccordionSummary`, and `AccordionDetails`. 

<PropsTable of={Accordion} />

### Accordion Item

The following props are available for the AccordionItem component:

<PropsTable of={AccordionItem} />

### Accordion Summary

The following props are available for the AccordionSummary component:

<PropsTable of={AccordionSummary} />

### Accordion Details

The following props are available for the AccordionDetails component:

<PropsTable of={AccordionDetails} />