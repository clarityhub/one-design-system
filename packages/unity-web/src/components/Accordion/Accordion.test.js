import React from 'react';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import renderer from 'react-test-renderer';

import Typography from '../Typography';
import Accordion from './Accordion';
import AccordionItem from './AccordionItem';
import AccordionSummary from './AccordionSummary';
import AccordionDetails from './AccordionDetails';

describe('<Accordion />', () => {
	it('renders with panels', () => {
		const node = (
			<Accordion>
				<AccordionItem>
					<AccordionSummary
						expandIcon={<ExpandMoreIcon />}
					>
						<Typography inline noMargin noPadding type="text" style={{ marginRight: '30px' }}><b>General settings</b></Typography>
						<Typography inline noMargin noPadding type="text">I am an expansion panel</Typography>
					</AccordionSummary>

					<AccordionDetails>
						<Typography>
                    Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget
                    maximus est, id dignissim quam.
						</Typography>
					</AccordionDetails>
				</AccordionItem>
				<AccordionItem>
					<AccordionSummary
						expandIcon={<ExpandMoreIcon />}
					>
						<Typography inline noMargin noPadding type="text" style={{ marginRight: '30px' }}><b>General settings</b></Typography>
						<Typography inline noMargin noPadding type="text">I am an expansion panel</Typography>
					</AccordionSummary>

					<AccordionDetails>
						<Typography>
                    Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget
                    maximus est, id dignissim quam.
						</Typography>
					</AccordionDetails>
				</AccordionItem>
				<AccordionItem>
					<AccordionSummary
						expandIcon={<ExpandMoreIcon />}
					>
						<Typography inline noMargin noPadding type="text" style={{ marginRight: '30px' }}><b>General settings</b></Typography>
						<Typography inline noMargin noPadding type="text">I am an expansion panel</Typography>
					</AccordionSummary>

					<AccordionDetails>
						<Typography>
                    Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget
                    maximus est, id dignissim quam.
						</Typography>
					</AccordionDetails>
				</AccordionItem>
			</Accordion>
		);

		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});