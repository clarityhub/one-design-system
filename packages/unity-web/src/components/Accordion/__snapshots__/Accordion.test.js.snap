// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders with panels 1`] = `
.emotion-13 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-clip: border-box;
  border-radius: 0.25rem;
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  background-color: #FFF;
  margin-bottom: 0;
  -webkit-transition: all 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: all 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  -webkit-transition-property: margin-top,margin-bottom;
  transition-property: margin-top,margin-bottom;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
}

.emotion-13 >:not(.eedi62r0) h3 {
  margin: 0;
  padding: 0;
}

.emotion-13 >:not(.eedi62r0) h4 {
  margin-top: 0.6rem;
}

.emotion-13 >:not(.eedi62r0) p:only-child {
  margin: 0;
}

.emotion-13:first-of-type {
  border-top-left-radius: 0.25rem !important;
  border-top-right-radius: 0.25rem !important;
}

.emotion-13:last-of-type {
  border-bottom-left-radius: 0.25rem !important;
  border-bottom-right-radius: 0.25rem !important;
}

.emotion-6 {
  background-color: transparent;
  border: 0;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  width: 100%;
  padding: 0.75rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-transition: padding 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: padding 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  text-align: left;
}

.emotion-0 {
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  color: #322439;
  padding: 0;
  margin-bottom: 0;
  margin-top: 0;
  display: inline-block;
}

.emotion-0 + .emotion-0 {
  margin-top: 1.6rem;
}

.emotion-0 strong,
.emotion-0 b {
  font-weight: 800;
}

.emotion-0 i,
.emotion-0 em {
  font-style: italic;
}

.emotion-0 + .emotion-0 {
  margin-bottom: 0;
  margin-top: 0;
}

.emotion-4 {
  display: inline-block;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  -webkit-transition: transform 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: transform 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.emotion-11 {
  overflow: hidden;
  -webkit-transition: max-height 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: max-height 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  max-height: 0;
}

.emotion-9 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: 1.25rem;
}

.emotion-8 {
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  color: #322439;
}

.emotion-8 + .emotion-8 {
  margin-top: 1.6rem;
}

.emotion-8 strong,
.emotion-8 b {
  font-weight: 800;
}

.emotion-8 i,
.emotion-8 em {
  font-style: italic;
}

<div
  className="emotion-45 emotion-46"
>
  <div
    className="emotion-13 emotion-14"
  >
    <button
      className="emotion-6 emotion-7"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="emotion-2 emotion-3"
      >
        <p
          className="emotion-0"
          style={
            Object {
              "marginRight": "30px",
            }
          }
        >
          <b>
            General settings
          </b>
        </p>
        <p
          className="emotion-0"
        >
          I am an expansion panel
        </p>
      </div>
      <div
        className="emotion-4 emotion-5"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-1"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </button>
    <div
      className="emotion-11 emotion-12"
      onClose={[Function]}
    >
      <div
        className="emotion-9 emotion-10"
      >
        <p
          className="emotion-8"
        >
          Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget maximus est, id dignissim quam.
        </p>
      </div>
    </div>
  </div>
  <div
    className="emotion-13 emotion-14"
  >
    <button
      className="emotion-6 emotion-7"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="emotion-2 emotion-3"
      >
        <p
          className="emotion-0"
          style={
            Object {
              "marginRight": "30px",
            }
          }
        >
          <b>
            General settings
          </b>
        </p>
        <p
          className="emotion-0"
        >
          I am an expansion panel
        </p>
      </div>
      <div
        className="emotion-4 emotion-5"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-1"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </button>
    <div
      className="emotion-11 emotion-12"
      onClose={[Function]}
    >
      <div
        className="emotion-9 emotion-10"
      >
        <p
          className="emotion-8"
        >
          Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget maximus est, id dignissim quam.
        </p>
      </div>
    </div>
  </div>
  <div
    className="emotion-13 emotion-14"
  >
    <button
      className="emotion-6 emotion-7"
      disabled={false}
      onClick={[Function]}
    >
      <div
        className="emotion-2 emotion-3"
      >
        <p
          className="emotion-0"
          style={
            Object {
              "marginRight": "30px",
            }
          }
        >
          <b>
            General settings
          </b>
        </p>
        <p
          className="emotion-0"
        >
          I am an expansion panel
        </p>
      </div>
      <div
        className="emotion-4 emotion-5"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-1"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </button>
    <div
      className="emotion-11 emotion-12"
      onClose={[Function]}
    >
      <div
        className="emotion-9 emotion-10"
      >
        <p
          className="emotion-8"
        >
          Nulla facilisi. Phasellus sollicitudin nulla et quam mattis feugiat. Aliquam eget maximus est, id dignissim quam.
        </p>
      </div>
    </div>
  </div>
</div>
`;
